extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -79,11 +79,11 @@ public ThreadContextController begin() {
                 // context active, store current state, start blank context anew and restore state afterwards
                 // it is not necessary to deactivate the context first - just overwrite the previous state
                 requestContext.activate();
-                return new RestoreContextController(requestContext, toRestore);
+                return new RestoreContextController(requestContext, toRestore, true);
             } else {
-                // context not active, activate blank one, deactivate afterwards
+                // context not active, activate blank one, destroy afterwards
                 requestContext.activate();
-                return requestContext::deactivate;
+                return requestContext::destroy;
```

Comment:
Shouldn't we return `requestContext::terminate` instead?

Reply:
Good point, that should have been `terminate` indeed; fixed now.